### PR TITLE
test: move to RTL all react-components/A*

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

our test use enzyme and we want to move to react 18

**What is the chosen solution to this problem?**

rewrite tests using RTL

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/ActionBar/ActionBar.test.js [hunk 1]
```diff
@@ -1,7 +1,58 @@
+/* eslint-disable react/prop-types */
 import React from 'react';
-import { mount } from 'enzyme';
+// rewrite thoses test with react-testing-library
```
Comment (Nigui) [inline packages/components/src/ActionBar/ActionBar.test.js:3]: Can be removed?

File: packages/components/src/ActionBar/ActionBar.test.js [hunk 2]
```diff
@@ -1,7 +1,58 @@
...
+		},
+		{
+			label: 'From Remote',
+			onClick: jest.fn(),
+		},
+	],
+	emptyDropdownLabel: 'No option',
+};
+
+const btnGroupAction = {
```
Comment (Nigui) [inline packages/components/src/ActionBar/ActionBar.test.js:39]: lint error (unused variable)

File: packages/components/src/ActionIntercom/Intercom.component.test.js [hunk 3]
```diff
@@ -30,6 +31,7 @@ describe('Intercom button', () => {
 	let insertionElement;
```
Comment (Nigui) [inline packages/components/src/ActionIntercom/Intercom.component.test.js:31]: looks not used anymore

File: packages/components/src/ActionList/ActionList.test.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import { Button } from '@talend/react-bootstrap';
 import { mount } from 'enzyme';
```
Comment (Nigui) [inline packages/components/src/ActionList/ActionList.test.js:2]: mount is unused

File: packages/components/src/ActionList/ActionList.test.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
+import { render, screen, configure } from '@testing-library/react';
```
Comment (Nigui) [inline packages/components/src/ActionList/ActionList.test.js:3]: configure is unused

File: packages/components/src/AppGuidedTour/AppGuidedTour.test.js [hunk 6]
```diff
@@ -1,9 +1,14 @@
 import React from 'react';
 import { act } from 'react-dom/test-utils';
 import { mount } from 'enzyme';
+// rewrite test using react-testing-library
```
Comment (Nigui) [inline packages/components/src/AppGuidedTour/AppGuidedTour.test.js:4]: to remove

File: packages/components/src/AppGuidedTour/AppGuidedTour.test.js [hunk 7]
```diff
@@ -19,18 +24,18 @@ describe('AppGuidedTour', () => {
...
-			wrapper.find('button[data-tour-elem="right-arrow"]').simulate('click');
-		});
+		// act(() => {
+		// 	wrapper.find('Toggle').prop('onChange')({ target: { checked: false } });
+		// });
+		await userEvent.click(screen.getByLabelText('Import demo content'));
+		await userEvent.click(screen.getByText('Let me try'));
+		// act(() => {
+		// 	wrapper.find('button[data-tour-elem="right-arrow"]').simulate('click');
+		// });
```
Comment (Nigui) [inline packages/components/src/AppGuidedTour/AppGuidedTour.test.js:38]: remove comments

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js [hunk 8]
```diff
@@ -135,47 +148,50 @@ describe('InjectDropdownMenuItem', () => {
...
-			<ActionDropdown
-				items={[{ label: 'item 1' }, { label: 'item 2' }]}
-				dropup={isInitialDropup}
-			/>,
-			{
-				attachTo: container,
-			},
+		// const container = screen.document.createElement('div');
+		// container.classList.add('tc-dropdown-container');
+		// container.getBoundingClientRect = () => containerPosition;
```
Comment (Nigui) [inline packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js:171]: to remove

**Review by Nigui — APPROVED** [review]
great work 👏 
Some cleaning comments only

**Review by romainseb — APPROVED** [review]

File: packages/components/src/ActionIntercom/Intercom.component.test.js (hunk 3 above)
Comment (jmfrancois) [reply packages/components/src/ActionIntercom/Intercom.component.test.js:31]: indeed

**Pull request merged**